### PR TITLE
Feature: metrics

### PR DESCRIPTION
**Pull request by obecny**
metrics

testing folder is for development / testing purposes.

the console.log from `proto.js` I will remove after review - might be easier to debug / test stuff

### CODE REVIEW

**Comment by obecny** [comment]
@mwear ^^

**Comment by mwear** [comment]
This all looks good to me, at least for an initial alpha. I have a larger question about what we want to do longer term. In this PR we've modified the metrics system that already ships as part of ls-trace-js: https://github.com/lightstep/ls-trace-js/pull/7/files#diff-ded06582bc11034346decd1b6b2bcbbb. 

Would we ever want the ability to choose between the existing dogstatsd implementation and the LS implementation? This is probably a question for @andrewhsu.

If so, would make sense copy [metrics.js](https://github.com/lightstep/ls-trace-js/pull/7/files#diff-ded06582bc11034346decd1b6b2bcbbb) and hook in a separate implementation that users can toggle between based on configuration. I'm not sure that we should do this, but I'm wondering if this is an approach we should consider.

File: packages/dd-trace/src/platform/node/metrics_lightstep.js [hunk 1]
```diff
@@ -0,0 +1,379 @@
+'use strict'
+
+// TODO: capture every second and flush every 10 seconds
```
Comment (mwear) [inline packages/dd-trace/src/platform/node/metrics_lightstep.js]: Is this comment taken care of by `_config.reportingInterval` now?

File: packages/dd-trace/src/platform/node/metrics_lightstep.js [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+
+const v8 = require('v8')
+const path = require('path')
+const os = require('os')
+// const Client = require('./dogstatsd');
```
Comment (mwear) [inline packages/dd-trace/src/platform/node/metrics_lightstep.js]: We probably don't need this.

File: testing/test.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+const express = require('express');
+const tracer = require('../packages/dd-trace').init({
+  clientToken: 'YOUR_TOKEN',
+  plugins: true,
+  runtimeMetrics: true,
+  reportingInterval: 5 * 1000,
+  componentName: 'ls-trace-js-testing',
+  // url: 'https://ingest.lightstep.com/metrics',
+  url: 'https://ingest.staging.lightstep.com/metrics',
```
Comment (mwear) [inline testing/test.js]: We probably want to the use the non-stating url when we merge this.

**Review by mwear — APPROVED** [review]
I had a couple of comments, but this looks great!

**Comment by obecny** [comment]
@mwear @andrewhsu 
I manage to fix all except tests for plugin `datadog-plugin-google-cloud-pubsub` and one test for `datadog-plugin-fs` I added there mark `// @TODO FIX_IT` , the rest are working fine

**Comment by mwear** [comment]
Thanks @obecny 👏 !

**Pull request merged**